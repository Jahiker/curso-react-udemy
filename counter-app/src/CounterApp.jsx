import React from "react";

import PropsType from "prop-types";

const handleAdd = (event) => {
  console.log("+1", event);
};

export const CounterApp = ({ value }) => {

  return (
    <>
      <h1>Counter App</h1>
      <h2> {value} </h2>
      <button onClick={handleAdd}>+1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropsType.number.isRequired,
};

CounterApp.defaultProps = {
  value: 0,
};
