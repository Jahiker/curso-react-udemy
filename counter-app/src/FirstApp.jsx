import React from 'react'
import PropTypes from 'prop-types'

export const FirstApp = ({ title, subtitle, number }) => {

  return (
    <>
        <h1>{ title }</h1>
        <p>{ subtitle }</p>
        <p>{ number }</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  number: PropTypes.number
}

FirstApp.defaultProps = {
  title: "Esto es el titulo",
  subtitle: "Esto es el subtitulo",
  number: 123
}