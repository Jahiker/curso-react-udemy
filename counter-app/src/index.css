html, body {
    background-color:  #21232a;
    color:  #fff;
    font-family: Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
    border-radius: 5px;
    min-width: 50px;
    height: 50px;
    font-weight: bold;
    border: none;
    margin-right: 5px;
}