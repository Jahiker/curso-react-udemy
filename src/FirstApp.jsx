import React from 'react'

const newMessage = {
    message: "Hola Mundo"
}

const description = () => "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ipsam optio eaque in sapiente quibusdam quia placeat iste unde harum, aspernatur culpa nesciunt officiis excepturi omnis perferendis quae aperiam. Quibusdam, facilis!"

export const FirstApp = () => {
  return (
    <>
        <h1>{ newMessage.message }</h1>
        <p>{ description() }</p>
    </>
  )
}
