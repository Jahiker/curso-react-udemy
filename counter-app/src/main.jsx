import React from "react";
import ReactDOM from "react-dom/client";

import "./index.css";

// import { CounterApp } from "./CounterApp";
import { FirstApp } from "./FirstApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <CounterApp value={10} /> */}
    <FirstApp title="Hola!" subtitle="Estamos testeando en React con Jest" />
  </React.StrictMode>
);
