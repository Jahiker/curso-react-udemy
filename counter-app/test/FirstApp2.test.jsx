import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Test on <FirstApp />", () => {
  const title = "Esto es el titulo";
  const subtitle = "Esto es el subtitulo";

  test("should match with snapshopt", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('should show the title "Esto es el titulo"', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('should show the title on h1 tag', () => {
    render(<FirstApp title={title} />);
    expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );
  })

  test('should show the prop subtitle', () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect( screen.getByText(subtitle) ).toBeTruthy();
    expect( screen.getAllByText(subtitle).length ).toBe(1);
  })
});
