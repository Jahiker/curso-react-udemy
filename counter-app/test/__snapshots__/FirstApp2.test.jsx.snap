// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <FirstApp /> should match with snapshopt 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Esto es el titulo
  </h1>
  <p>
    Esto es el subtitulo
  </p>
  <p>
    123
  </p>
</div>
`;
